openhabian-setup.sh

```diff
@@ -108,6 +108,8 @@ if [[ -n "$UNATTENDED" ]]; then
   clean_config_userpw
   frontail_setup
   zram_setup
+  amanda_setup
```

Comment:
Remove, as it is not necessary.

Reply:
it is! will return if $backupdrive unset

Reply:
But it does not actaully setup the amanda backup? right?

Reply:
yes it does, with default params

Reply:
But then it would be overwritten by the mirror SD which creates another amanda config, right?

Reply:
you're right, I removed that call in setup_mirror_SD

Reply:
I would actually prefer it to take place there, it makes incompatibility with #1055 much simpler.

Reply:
I understand but that sort of nesting is a no-go to me from a SW architecture POV.
All install routines should be capable of being called directly and without dependencies on each other, for both, unattended and interactive install, and handle interativeness (or not) themselves.

Reply:
It's not really nesting. The create Amanda config function is intended for that purpose and should function just fine. Whereas the Amanda setup function should be used for an interactive setup.

Reply:
No it's insufficient to create the config only, and the code for the missing parts exists in amanda_setup() and used to be duplicated in mirror_SD. Not desirable from the SW arch perspective either.

Reply:
SW Arch?

Reply:
software architecture
Ok a big word for little code like this but still valid IMHO

Reply:
Very well, however for a more object oriented design, it might be better to create a new function to install and setup necessary Amanda components (ie `amanda_install`). And then keep `amanda_setup` as a interactive configuration setup and use those to create a more modular and simpler design.

Reply:
yeah but bash programming isn't OO, pure imperative procedural that is, and I don't want to start mixing styles either.

Reply:
But in this case I think it works out to a good compromise and allows for much more modular code.

Reply:
In the first place, I'm happy to have it working and I don't see the point in (more) modularization there so I don't want to refactor that upfront ... but I'm ok with when you change it after the initial merge as part of your ongoing standardization efforts. Are you ok with that, too ?

Reply:
Fair enough.